### PR TITLE
Contracts monorepo 7

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

Refactors out extensions and examples. Now the core package has only protocol contracts. And is renamed to protocol.

## Types of changes

* New feature (non-breaking change which adds functionality)

* Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [x] Prefix PR title with `[WIP]` if necessary.
*   [x] Add tests to cover changes as needed.
*   [x] Update documentation as needed.
*   [x] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/extensions/package.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        "lint": "tslint --format stylish --project . --exclude ./generated-wrappers/**/* --exclude ./generated-artifacts/**/* --exclude **/lib/**/* && yarn lint-contracts",
+        "coverage:report:text": "istanbul report text",
+        "coverage:report:html": "istanbul report html && open coverage/index.html",
+        "profiler:report:html": "istanbul report html && open coverage/index.html",
+        "coverage:report:lcov": "istanbul report lcov",
+        "test:circleci": "yarn test",
+        "lint-contracts": "solhint -c ../.solhint.json contracts/**/**/**/**/*.sol"
+    },
+    "config": {
+        "abis": "generated-artifacts/@(DutchAuction|Forwarder|OrderValidator).json"
```
Comment (fabioberger) [inline contracts/extensions/package.json]: Hm, `OrderValidator` isn't really an "extension". It doesn't sit in-front of the `exchange` contract and act as a `sender` for orders. I think we should move it to `exchange`, since it'll become the helper function contract that we are going to ship with 2.1, and will become an integral part of our system of smart contracts offering. Thoughts @abandeali1?

File: packages/instant/package.json [hunk 2]
```diff
@@ -96,6 +96,6 @@
         "webpack-dev-server": "^3.1.9"
     },
     "publishConfig": {
-        "access": "private"
+        "access": "public"
```
Comment (fabioberger) [inline packages/instant/package.json]: Why was this changed?

File: .circleci/config.yml [hunk 3]
```diff
@@ -44,7 +44,8 @@ jobs:
             - run: yarn wsrun test:circleci @0x/contracts-utils
             - run: yarn wsrun test:circleci @0x/contracts-libs
             - run: yarn wsrun test:circleci @0x/contracts-tokens
-            - run: yarn wsrun test:circleci @0x/contracts-core
+            - run: yarn wsrun test:circleci @0x/contracts-extensions
+            - run: yarn wsrun test:circleci @0x/contracts-protocol
```
[the remark this answers is not in the document]
Comment (LogvinovLeon) [reply .circleci/config.yml]: They don't have any tests

File: packages/instant/package.json (hunk 2 above)
Comment (LogvinovLeon) [reply packages/instant/package.json]: Good catch. Should revert
